flare/model_zoo/distributional_rl_models.py

```diff
@@ -0,0 +1,72 @@
+from simple_models import SimpleModelQ
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import math
+
+
+class SimpleModelC51(SimpleModelQ):
+    def __init__(self, dims, num_actions, mlp, vmax, vmin, bins):
+        super(SimpleModelC51, self).__init__(dims, num_actions, mlp)
+        assert bins > 1
+        assert vmax > vmin
+        self.vmax = vmax
+        self.vmin = vmin
+        self.bins = bins
+
+        self.delta_z = float(self.vmax - self.vmin) / (self.bins - 1)
+        atoms = [vmin + i * self.delta_z for i in xrange(self.bins)]
+        self.atoms = torch.tensor(atoms)
+
+    def value(self, inputs, states):
+        q_distributions = self.mlp(inputs.values()[0])
+        q_distributions = q_distributions.view(-1, self.num_actions, self.bins)
+        q_distributions = F.softmax(q_distributions, 2)
+        q_values = torch.matmul(q_distributions, self.atoms)
+        return dict(q_value=q_values, q_value_list=q_distributions), states
```

Comment:
q_value_list -> q_value_distribution; do the same for all the models in this file